src/details/ArborX_DetailsTreeConstruction.hpp

```diff
@@ -209,20 +136,42 @@ namespace
 int constexpr UNTOUCHED_NODE = -1;
 } // namespace
 
-template <typename MemorySpace>
-class GenerateHierarchyFunctor
+template <typename ExecutionSpace, typename Primitives, typename Nodes>
+inline void initializeSingleLeafNode(ExecutionSpace const &space,
+                                     Primitives const &primitives,
+                                     Nodes const &leaf_nodes)
+{
+  using Access = AccessTraits<Primitives, PrimitivesTag>;
+
+  Kokkos::parallel_for(
+      "ArborX::BVH::BVH::initialize_single_leaf",
+      Kokkos::RangePolicy<ExecutionSpace>(space, 0, 1), KOKKOS_LAMBDA(int i) {
+        Box bbox{};
+        expand(bbox, Access::get(primitives, 0));
+        leaf_nodes(0) = Details::makeLeafNode(0, std::move(bbox));
+      });
+}
+
+template <typename Primitives, typename MemorySpace>
```

Comment:
Would prefer the opposite order.

Reply:
I wanted to change virtually everything about this class and you convinced me not to...